_includes/social.html

```diff
@@ -1,34 +1,20 @@
 <!-- Meta -->
-<meta name="description" content="{{ site.description }}">
+<meta name="description" content="{{ page.description | default: site.description }}">
 <meta name="author" content="{{ site.authors }}">
 
 <!-- Twitter -->
+<meta name="twitter:card" content="{% if page.title %}summary{% else %}summary_large_image{% endif %}">
 <meta name="twitter:site" content="@{{ site.twitter }}">
 <meta name="twitter:creator" content="@{{ site.twitter }}">
-
-{% if page.title %}
-  <meta name="twitter:card" content="summary">
-  <meta name="twitter:title" content="{{ page.title }}">
-  <meta name="twitter:description" content="{{ page.description }}">
-  <meta name="twitter:image" content="{{ site.url }}{{ site.social_logo_path }}">
-{% else %}
-  <meta name="twitter:card" content="summary_large_image">
-  <meta name="twitter:title" content="{{ site.title }}">
-  <meta name="twitter:description" content="{{ site.description }}">
-  <meta name="twitter:image" content="{{ site.url }}{{ site.social_image_path }}">
-{% endif %}
+<meta name="twitter:title" content="{{ page.title | default: site.title }}">
+<meta name="twitter:description" content="{{ page.description | default: site.description }}">
+<meta name="twitter:image" content="{{ site.url }}{{ site.social_image_path }}">
 
 <!-- Facebook -->
-{% if page.title %}
-  <meta property="og:url" content="{{ site.url }}{{ page.url }}">
-  <meta property="og:title" content="{{ page.title }}">
-  <meta property="og:description" content="{{ page.description }}">
-  <meta property="og:type" content="website">
-{% else %}
-  <meta property="og:url" content="{{ site.url }}">
-  <meta property="og:title" content="{{ site.title }}">
-  <meta property="og:description" content="{{ site.description }}">
-{% endif %}
+<meta property="og:url" content="{% if page.title %}{{ site.url }}{{ page.url }}{% else %}{{ site.url | append: "/" }}{% endif %}">
```

Comment:
This might be possible to simplify more by just using `{{ site.url }}{{ page.url }}` since for the index page